app/src/main/java/org/mozilla/focus/download/DownloadInfoManager.java

```diff
@@ -101,21 +101,31 @@ protected void onUpdateComplete(int token, Object cookie, int result) {
         }
 
         @Override
-        protected void onQueryComplete(int token, Object cookie, Cursor cursor) {
+        protected void onQueryComplete(int token, final Object cookie, final Cursor cursor) {
             switch (token) {
                 case TOKEN:
-                    if (cookie != null) {
-                        List<DownloadInfo> downloadInfoList = new ArrayList<>();
-                        if (cursor != null) {
-                            while (cursor.moveToNext()) {
-                                final DownloadInfo downloadInfo = cursorToDownloadInfo(cursor);
-                                downloadInfoList.add(downloadInfo);
+                    ThreadUtils.postToBackgroundThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            if (cookie != null) {
```

Comment:
We didn't close the cursor when the cookie is null, right?

Reply:
Fixed. Thanks!